tests/communication_1to1_test.py

```diff
@@ -50,47 +54,40 @@ def execute(self):
 class SubscriberOp(Op):
     def __init__(self, name):
         super(SubscriberOp, self).__init__(name)
+        self._cnt = 0
 
     @staticmethod
     def setup_streams(input_streams):
         input_streams.add_callback(SubscriberOp.on_msg)
         return [DataStream(data_type=String, name='sub_out')]
 
     def on_msg(self, msg):
-        self.get_output_stream('sub_out').send(msg)
+        if type(msg) is not str:
+            data = msg.data
+        count = int(data.split(" ")[1])
+        assert count < MAXIMUM_COUNT and count == self._cnt, \
+            'received count %d should be equal to %d and smaller than maximum count %d' \
+            % (count, self._cnt, MAXIMUM_COUNT)
+        self._cnt += 1
         print('%s received %s' % (self.name, msg))
 
     def execute(self):
         self.spin()
```

Comment:
Here instead of spinning we could:
```
while self._cnt < MAX_COUNT:
  time.sleep(0.1)
```
We would also have to set a timeout so that the test completes with an error in the case when we don't receive any messages.